src/pages/studyView/table/FixedHeaderTable.tsx

```diff
@@ -0,0 +1,161 @@
+    Table as RVTable,
+    TableCellProps
+} from 'react-virtualized';
+import 'react-virtualized/styles.css';
+import {action, observable} from "mobx";
+import styles from "./tables.module.scss";
+import * as _ from 'lodash';
+import {observer} from "mobx-react";
+import classnames from 'classnames';
+import {If} from 'react-if';
+import {bind} from "bind-decorator";
+
+export type IFixedHeaderTableProps<T> = {
+    columns: Column<T>[],
+    data: T[];
+    sortBy?: string;
+    width?: number;
+    height?: number;
+    // This property is just used to force to rerender the table when selectedGenes is modified.
+    // We don't actually use the property anywhere in the table.
+    selectedGenes?: string[];
+    selectedRows?: number[];
+    showSelectSamples?: boolean;
+    afterSelectingRows?: () => void;
+};
+
+const RVSDTtoStrType = {
+    ['desc' as SortDirection]: RVSortDirection.DESC,
+    ['asc' as SortDirection]: RVSortDirection.ASC
+};
+
+@observer
+export default class FixedHeaderTable<T> extends React.Component<IFixedHeaderTableProps<T>, {}> {
+    private _store: LazyMobXTableStore<T>;
+    @observable private _sortBy: string;
+    @observable private _sortDirection: SortDirection;
+
+    public static defaultProps = {
+        showSelectSamples: false,
+        selectedRows: [],
+        selectedGenes: [],
+        width : 398,
+        height: 350,
+        sortBy: ''
+    };
+
+    constructor(props: IFixedHeaderTableProps<T>) {
+        super(props);
+        this._sortBy = props.sortBy!;
+        const sortByColumn = _.find(this.props.columns, column => column.name === this._sortBy);
+        this._sortDirection = _.isUndefined(sortByColumn) ? 'desc' as 'desc' : sortByColumn.defaultSortDirection || 'desc' as 'desc';
+
+        this.initDataStore();
+    }
+
+    @action
+    initDataStore() {
+        this._store = new LazyMobXTableStore<T>({
+            columns: this.props.columns,
+            data: this.props.data,
+            initialSortColumn: this._sortBy,
+            initialSortDirection: this._sortDirection
+        });
+    }
+
+    @bind
+    rowClassName({index}: any) {
+        if (_.includes(this.props.selectedRows, index)) {
+            return styles.highlightedRow;
+        } else if (index < 0) {
+            return styles.headerRow;
+        } else {
+            return index % 2 === 0 ? styles.evenRow : styles.oddRow;
+        }
+    }
+
+    @bind
+    rowGetter({index}: any) {
+        return this._store.dataStore.sortedFilteredData[index];
+    }
+
+    @bind
+    getColumn(columnKey: string) {
+        return _.keyBy(this.props.columns, column => column.name)[columnKey];
+    }
+
+    @bind
+    @action
+    sort({sortBy}: any) {
+        this._store.defaultHeaderClick(this.getColumn(sortBy));
+        this._sortBy = sortBy;
+        this._sortDirection = this._store.dataStore.sortAscending ? 'asc' as 'asc' : 'desc' as 'desc';
+    }
+
+    @bind
+    @action
+    onFilterTextChange() {
```

Comment:
I think this code is very similar (if not identical) to the one we have in `LazyMobXTable`, it would be great to have a shared utility function for this.